packages/core/src/components/axes/axis.ts

```diff
@@ -205,6 +205,10 @@ export class Axis extends Component {
 			scale.domain(zoomDomain);
 		}
 
+		if (!Tools.getProperty(options, "axes", axisPosition, "visible")) {
+			return;
+		}
```

Comment:
This is great, but we're still running logic that doesn't need running when we've `return`ed.

- `tickHeight`
- `fakeTick`
- `truncationNumCharacter` and all the truncation-related info
- and more

Please do a sweep above the if-statement and move below the if-statement anything that we wouldn't need if we return because of the `visible` flag.

Also, we need to add `.attr("aria-hidden", true)` to the axis ref if it's not visible.

Ideally we also wouldn't create a 2nd `invisible` axis ref and just use the original one as the invisible ref since it is exactly that

Reply:
we've still got all of these things left to do^^